programs/pkey/gen_key.c

```diff
@@ -281,8 +298,15 @@ int main( int argc, char *argv[] )
                 opt.format = FORMAT_PEM;
             else if( strcmp( q, "der" ) == 0 )
                 opt.format = FORMAT_DER;
+            else
+                goto usage;
+        }
+        else if( strcmp( p, "syntax" ) == 0 )
+        {
+            if( strcmp( p, "pkcs1") == 0 )
```

Comment:
This should test `q` rather than `p`. Maybe the tests could be extended to also use this option? Currently, they only use pkcs8 format, which is why this was not caught.